app/views/component/search/_category.html.haml

```diff
@@ -81,10 +84,7 @@
       $('#main_category').val('');
     }
 
-    $('#form-search').on('submit', function() {
+    $('#keyword-search-button').on('submit', function() {
       var keyword = $('#keyword-search-box').val();
-      if (keyword && !#{@matched_category.present?}) {
-        clearSubcategories();
-      }
```

Comment:
This code was causing the category selections to disappear when applying filters, if no matching category had been found.